content/en/data_streams/messages.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Messages
+---
+
+The Messages feature allows you to view messages at specific offsets within Kafka. Inspecting these messages can be crucial when troubleshooting issues related to specific services or data streams.
+
+## Setup
+Before diving into detailed configuration steps, here's a quick overview of what's involved in setting up the Messages feature:
+
+| Step               | Description                                                                                                                                       | Setup Guides              |
+|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------|
+| Agent setup        | An agent needs Kafka consumer integration enabled, be on at least version 7.69.0, and have remote configuration enabled                           | [Agent Setup][1]          |
+| Verify permissions | You'll need a few log access permissions which usually come standard as well as `Data Streams Monitoring Capture Messages` enabled in Datadog     | [Required Permissions][2] |
+
+
+### Agent Setup
+
+#### Selecting an Agent
+Any Datadog agent having access to Kafka can be used for setup.
+
+If you self-host Kafka, set up Messages on your Kafka broker's agent. Otherwise, choose any agent your producer or consumer services communicate with that has access to your Kafka cluster.
+
+#### Step-by-step guide
+
+Complete the following steps on the same Datadog agent:
+
+##### 1. Verify agent version
+
+Ensure your agent is running version `7.69.0` or later.
+
+##### 2. Enable [remote configuration][3]
+
+Confirm Remote Configuration is enabled for your agent (typically enabled by default). Verify this setting on the [Fleet Automation page][4] and also confirm it at the organizational level via the [Remote Configuration settings page][5].
+
+##### 3. Configure Kafka consumer integration
+Create or update the Kafka consumer configuration file at
+
+```
+[agent config directory]/kafka_consumer.d/conf.yaml
+```
+
+Here's an example configuration (see a detailed sample [here][9]):
+```yaml
+init_config:
+
+instances:
+  - kafka_connect_str: kafka:29092 # Replace with your Kafka cluster address
+    monitor_unlisted_consumer_groups: true
+    tags:
+      - env:staging
```

Comment:
The metrics inherit automatically any host tags of the host the agent is running on. This will include environment most of the time if the setup of Datadog is done well.

Ideally, we would not use `env` when filtering the Kafka lag metric, but use `kafka_cluster_id` instead.

Reply:
Ohh good to know. I'll modify the wording that the tags are inherited

We don't use `env` in the remote config but we might need it in DSM when filtering for broker offset metric when users are filtering on a datacenter or some other primary tag

Reply:
If we know the kafka_cluster_id of the topic side panel, we could filter on the broker offset metric using only kafka_cluster_id instead of using `env` or primary tags. It would be more reliable since the `kafka_cluster_id` is automatically added.